backend/src/bundles/statistics/statistics.service.ts

```diff
@@ -0,0 +1,178 @@
+import {
+    addDays,
+    addMonths,
+    addWeeks,
+    format,
+    isSameDay,
+    min,
+    subDays,
+    subMonths,
+    subWeeks,
+} from 'date-fns';
+import {
+    type StatisticsRecord,
+    compareDateInDiapasonWithoutTime,
+    StatisticsPeriods,
+} from 'shared/build/index.js';
+
+import { type ResumeShareService } from '../resume-share/resume-share.service.js';
+import {
+    type GetStatisticsResponseDto,
+    type ResumeShareDetailsGetResponseDto,
+} from './types/types.js';
+
+class StatisticsService {
+    private resumeShareService: ResumeShareService;
+
+    public constructor(resumeShareService: ResumeShareService) {
+        this.resumeShareService = resumeShareService;
+    }
+
+    private getStatisticsHandleWeek(
+        data: ResumeShareDetailsGetResponseDto[],
+    ): GetStatisticsResponseDto {
+        const statistics: StatisticsRecord[] = [];
+
+        const dateWeekAgo = addDays(subWeeks(new Date(), 1), 1);
+
+        const accesses = data.flatMap((resume) => resume.accesses);
+
+        const currentDate = new Date();
+
+        for (let day = dateWeekAgo; day <= currentDate; day = addDays(day, 1)) {
+            statistics.push([
+                format(day, 'MMM d'),
+                accesses.filter((access) => {
+                    return isSameDay(
+                        new Date(access.resumeShareAccessTime),
+                        day,
+                    );
+                }).length,
+            ]);
+        }
+
+        return {
+            data: statistics,
+            sum: statistics.reduce(
+                (accumulator, currentValue) => accumulator + currentValue[1],
+                0,
+            ),
+        };
+    }
+
+    private getStatisticsHandleMonth(
+        data: ResumeShareDetailsGetResponseDto[],
+    ): GetStatisticsResponseDto {
+        const statistics: StatisticsRecord[] = [];
+
+        const dateMonthAgo = subMonths(new Date(), 1);
+
+        const accesses = data.flatMap((resume) => resume.accesses);
+
+        const currentDate = new Date();
+
+        for (
+            let day = dateMonthAgo;
+            day <= currentDate;
+            day = addWeeks(day, 1)
+        ) {
+            const dayInAWeek = subDays(addWeeks(day, 1), 1);
+
+            const dayLimit = min([dayInAWeek, new Date()]);
+
+            const stringDate = `${format(day, 'MMM d')} - ${format(
+                dayLimit,
+                'MMM d',
+            )}`;
+
+            statistics.push([
+                stringDate,
+                accesses.filter((access) => {
+                    return compareDateInDiapasonWithoutTime(
+                        new Date(access.resumeShareAccessTime),
+                        day,
+                        dayLimit,
+                    );
+                }).length,
+            ]);
+        }
+
+        return {
+            data: statistics,
+            sum: statistics.reduce(
```

Comment:
This `reduce` is getting used 3 times, not DRY - I would write a helper
also, what is `sum`, sum of what? it would be more easier for other devs to read
also, why reduce? can it be replaced? 

` const totalDailyAccesses = statistics.reduce((accumulator, currentValue) => accumulator + currentValue[1], 0);
`